# Estimation And Confidence Intervals

In quality control processes, especially when dealing with high-value items, destructive sampling is a necessary but costly method to ensure product quality. The test to determine whether an item meets the quality standards destroys the item, leading to the requirement of small sample sizes due to cost constraints.

**Scenario**

---


A manufacturer of print-heads for personal computers is interested in estimating the mean durability of their print-heads in terms of the number of characters printed before failure. To assess this, the manufacturer conducts a study on a small sample of print-heads due to the destructive nature of the testing process.

In [ ]:
import numpy as np
import scipy.stats as stats #imports the statistical functions from SciPy and allows them to be called using the short alias stats.
#it  is a Python library built on top of NumPy that provides a massive collection of algorithms and high-level tools for scientific and technical computing.

data = np.array([1.13, 1.55, 1.43, 0.92, 1.25, 1.36, 1.32, 0.85, 1.07, 1.48, 1.20, 1.33, 1.18, 1.22, 1.29])

n = len(data)
mean = np.mean(data)
s = np.std(data, ddof=1)  # sample standard deviation
#The parameter ddof=1 stands for Delta Degrees of Freedom equals 1.
#It is a crucial argument used in the numpy.std() (standard deviation) and numpy.var() (variance) functions to ensure the calculation is based on the sample rather than the entire population

mean, s


(np.float64(1.2386666666666666), np.float64(0.19316412956959936))

# a. Build 99% Confidence Interval Using Sample Standard Deviation:
Assuming the sample is representative of the population, construct a 99% confidence interval for the mean number of characters printed before the print-head fails using the sample standard deviation. Explain the steps you take and the rationale behind using the t-distribution for this task.

In [ ]:
#The confidence interval for the population mean when the population standard deviation is unknown is:CI=xˉ±tα/2, df×sn
alpha = 0.01 #Confidence level=1−α=0.99
df = n - 1 #degrees of freedom for t-distribution.
t_critical = stats.t.ppf(1 - alpha/2, df)
#This gets the t-value that cuts off the upper 0.5%
margin_error = t_critical * (s / np.sqrt(n)) #how much above and below the sample mean the true mean might lie.
lower = mean - margin_error
upper = mean + margin_error
print("99% Confidence Interval using t-distribution:")
print(f"Mean = {mean:.4f}, s = {s:.4f}, df = {df}")
print(f"t-critical = {t_critical:.4f}")
print(f"CI = ({lower:.4f}, {upper:.4f})\n")

99% Confidence Interval (using t-distribution):
Mean = 1.2387, s = 0.1932, df = 14
t-critical = 2.9768
CI = (1.0902, 1.3871)



**b. Build 99% Confidence Interval Using Known Population Standard Deviation**

---


If it were known that the population standard deviation is 0.2 million characters, construct a 99% confidence interval for the mean number of characters printed before failure.

In [ ]:
sigma = 0.2 #population standard deviation
z_critical = stats.norm.ppf(1 - alpha/2) #stats.norm.ppf() finds the z-critical value (the cutoff point on the standard normal distribution).
#1 - alpha/2 = 1 - 0.005 = 0.995
margin_error_z = z_critical * (sigma / np.sqrt(n))
lower_z = mean - margin_error_z
upper_z = mean + margin_error_z

print("99% Confidence Interval (using z-distribution):")
print(f"Known sigma = {sigma}")
print(f"z-critical = {z_critical:.4f}")
print(f"CI = ({lower_z:.4f}, {upper_z:.4f})")

99% Confidence Interval (using z-distribution):
Known sigma = 0.2
z-critical = 2.5758
CI = (1.1057, 1.3717)


**(a):**

---



CI=1.239±2.977×0.19315=(1.090,1.387)

99% confident the true mean durability lies in this range.

---



**(b) :**

CI=1.239±2.576×0.215=(1.106,1.372)

99% confident the true mean durability lies in this range.

# Hypothesis Testing

Hypothesis testing is a statistical method used to determine if there is enough evidence in a sample of data to reject a specific assumption, called the null hypothesis, about a population parameter. It's a formal procedure for using sample data to draw conclusions about a population, quantifying the likelihood that an observed pattern or relationship occurred by chance.

Bombay hospitality Ltd. operates a franchise model for producing exotic Norwegian dinners throughout New England. The operating cost for a franchise in a week (W) is given by the equation W = $1,000 + $5X, where X represents the number of units produced in a week. Recent feedback from restaurant owners suggests that this cost model may no longer be accurate, as their observed weekly operating costs are higher.

To investigate the restaurant owners' claim about the increase in weekly operating costs using hypothesis testing.

# 1. State the Hypotheses statement:

First, we need to calculate the theoretical mean weekly cost (μ) based on the model and the given mean production (X):\
μ=W=1,000+5X\
Given X=600 units:\
μ=1,000+5(600)=1,000+3,000=4,000\
The claim is that the true mean weekly cost is higher than 4,000.\

---


* **Null Hypothesis (H0 ):** The actual mean weekly operating cost is equal to or less than the theoretical cost.\
H0 :μ≤4,000.\

---


* **Alternative Hypothesis (Ha ):** The actual mean weekly operating cost is higher than the theoretical cost.\
Ha :μ>4,000\

**This is a one-tailed (right-tailed) test.**

In [5]:
import math
import scipy.stats as stats
#Given data
sample_mean = 3050
theoritical_mean = 1000 + 5 * 600
sigma = 5 * 25
n = 25
#calculate the test statistic
standard_error = sigma / math.sqrt(n)
z_value = (sample_mean - theoritical_mean) / standard_error
z_value
print(f"The test statistic (Z) is: {z_value}")
#critical value for one-tailed test at alpha = 0.05
alpha = 0.05 #3. Determine the Probability and compare
z_critical = stats.norm.ppf(1 - alpha)
z_critical
print(f"\nThe critical value (Z) at alpha = {alpha} is: {z_critical}")
print(f"\nsample_mean: {sample_mean}")
print(f"\ntheoritical_mean: {theoritical_mean}")
print(f"\nstandard_error: {standard_error}")
print(f"\nz_value: {z_value}")
print(f"\nz_critical: {z_critical}")


The test statistic (Z) is: -38.0

The critical value (Z) at alpha = 0.05 is: 1.6448536269514722

sample_mean: 3050

theoritical_mean: 4000

standard_error: 25.0

z_value: -38.0

z_critical: 1.6448536269514722


# 4. Make a Decision:

In [6]:
if z_value < z_critical:
    print("reject the null hypothesis")
else:
    print("do not reject the null hypothesis")

reject the null hypothesis


# 5. Conclusion:

In [7]:
print("Based on the decision, there is no strong evidence to support the claim that the weekly operating costs are higher than the model suggests.")

Based on the decision, there is no strong evidence to support the claim that the weekly operating costs are higher than the model suggests.
